subprojects/worker-processes/src/main/java/org/gradle/process/internal/worker/messaging/WorkerConfigSerializer.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.process.internal.worker.messaging;
+
+import org.gradle.api.Action;
+import org.gradle.api.logging.LogLevel;
+import org.gradle.internal.io.ClassLoaderObjectInputStream;
+import org.gradle.internal.remote.internal.inet.MultiChoiceAddress;
+import org.gradle.internal.remote.internal.inet.MultiChoiceAddressSerializer;
+import org.gradle.internal.serialize.Decoder;
+import org.gradle.internal.serialize.Encoder;
+import org.gradle.internal.serialize.InputStreamBackedDecoder;
+import org.gradle.internal.serialize.OutputStreamBackedEncoder;
+import org.gradle.internal.serialize.Serializer;
+import org.gradle.process.internal.worker.WorkerProcessContext;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+/**
+ * Serializes and de-serializes {@link WorkerConfig}s.
+ */
+public class WorkerConfigSerializer implements Serializer<WorkerConfig> {
+
+    public static final WorkerConfigSerializer INSTANCE = new WorkerConfigSerializer();
+    private WorkerConfigSerializer() {
+        // Private to enforce singleton.
+    }
+
+    public void writeTo(WorkerConfig config, DataOutputStream dos) throws IOException {
+        OutputStreamBackedEncoder encoder = new OutputStreamBackedEncoder(dos);
+        new WorkerConfigSerializer().write(encoder, config);
+        encoder.flush();
```

Comment:
Does this flush need to be in a `finally`?

Reply:
The prior code did not but I think it would make sense to have it.